config/config.go

```diff
@@ -115,9 +135,23 @@ func (c Configuration) New(
 	serviceName string,
 	options ...Option,
 ) (opentracing.Tracer, io.Closer, error) {
-	if serviceName == "" {
+	if serviceName == "" && c.ServiceName == "" {
```

Comment:
I think this check is not necessary, since it will be done by NewTracer anyway

Reply:
True, but given that we have `ServiceName` as part of the `Configuration` object, this would cover the following scenario:

```golang
cfg := &Configuration{ServiceName: "my-service", ...}
_, closer, err := cfg.New("")
```
Without the right side of the check, the `ServiceName` from the `Configuration` is ignored and the error is returned ("no service name provided").

Reply:
not sure I follow. If serviceName arg is not blank, it will override c.ServiceName, otherwise it wouldn't. In either case it will be followed by a call to `c.NewTracer()` which will return an error if `c.ServiceName` is blank.

Reply:
If you call `NewTracer()`, yes. But `New("")` might be called directly:

```golang
cfg := &Configuration{ServiceName: "my-service", ...}
_, closer, err := cfg.New("")
```

Reply:
but New() delegates to NewTracer():
```
return c.NewTracer(options...)
```

Reply:
Oh, I see now what you mean. Agree :)